controllers/hetznercluster_controller.go

```diff
@@ -500,73 +505,101 @@ func reconcileTargetSecret(ctx context.Context, clusterScope *scope.ClusterScope
 	if err != nil {
 		return reconcile.Result{}, fmt.Errorf("failed to get client: %w", err)
 	}
 
-	secret := &corev1.Secret{
+	// To ensure compatibilty with both CCMs, create always a secret with name "hcloud" in the
+	// wl-cluster.
+	names := []string{clusterScope.HetznerCluster.Spec.HetznerSecret.Name}
+	if clusterScope.HetznerCluster.Spec.HetznerSecret.Name != "hcloud" {
+		names = append(names, "hcloud")
+	}
+
+	for _, name := range names {
+		err = reconcileOneWorkloadClusterSecret(ctx, clusterScope, wlClient, name)
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("failed to reconcile wl-cluster secret %q: %w",
+				name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func reconcileOneWorkloadClusterSecret(ctx context.Context, clusterScope *scope.ClusterScope, wlClient client.Client, name string) error {
+	wlSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      clusterScope.HetznerCluster.Spec.HetznerSecret.Name,
+			Name:      name,
 			Namespace: metav1.NamespaceSystem,
 		},
 	}
 
 	// Make sure secret exists and has the expected values
-	_, err = controllerutil.CreateOrUpdate(ctx, client, secret, func() error {
-		tokenSecretName := types.NamespacedName{
+	_, err := controllerutil.CreateOrUpdate(ctx, wlClient, wlSecret, func() error {
+		mgtSecretName := types.NamespacedName{
 			Namespace: clusterScope.HetznerCluster.Namespace,
 			Name:      clusterScope.HetznerCluster.Spec.HetznerSecret.Name,
 		}
 		secretManager := secretutil.NewSecretManager(clusterScope.Logger, clusterScope.Client, clusterScope.APIReader)
-		tokenSecret, err := secretManager.AcquireSecret(ctx, tokenSecretName, clusterScope.HetznerCluster, false, clusterScope.HetznerCluster.DeletionTimestamp.IsZero())
+		mgtSecret, err := secretManager.AcquireSecret(ctx, mgtSecretName, clusterScope.HetznerCluster, false, clusterScope.HetznerCluster.DeletionTimestamp.IsZero())
 		if err != nil {
 			return fmt.Errorf("failed to acquire secret: %w", err)
 		}
 
-		hetznerToken, keyExists := tokenSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken]
+		hcloudToken, keyExists := mgtSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken]
 		if !keyExists {
 			return fmt.Errorf("error key %s does not exist in secret/%s: %w",
 				clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken,
-				tokenSecretName,
+				mgtSecretName,
 				err,
 			)
 		}
 
-		if secret.Data == nil {
-			secret.Data = make(map[string][]byte)
+		if wlSecret.Data == nil {
+			wlSecret.Data = make(map[string][]byte)
 		}
 
-		secret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken] = hetznerToken
+		wlSecret.Data[clusterScope.HetznerCluster.Spec.HetznerSecret.Key.HCloudToken] = hcloudToken
+
+		// upstream hcloud-ccm uses the secret key "token", while the old Syself ccm used "hcloud".
+		// For compatibilty, we create always the other key, too.
+		for _, key := range []string{"token", "hcloud"} {
```

Comment:
right now we create a duplicate right? Wouldn't it be enough to create the right key (e.g. "token" for the upstream ccm)?

Reply:
So what you currently do is storing the same hcloud token three times in the same secret, right? 
1. using the configurable key from HetznerCluster.Spec.HetznerSecret.Key.HCloudToken
2. hardcoded key "token"
3. hardcoded key "hcloud".

Right? 

I think this will confuse people. We should check what is really necessary and then make the documentation clear. Otherwise we will have to start documenting why we have the same thing thrice in the secret.

Reply:
I suppose so. Can you discuss this with @batistein?

Reply:
@janiskemper I talked to Sven. He is fine with creating three keys in the secret. But he does not want to deprecate this setting.